Source/JavaScriptCore/bytecode/SpeculatedType.cpp

```diff
@@ -855,7 +855,9 @@ SpeculatedType typeOfDoubleNegation(SpeculatedType value)
 {
     // Changing bits can make pure NaN impure and vice versa:
     // 0xefff000000000000 (pure) - 0xffff000000000000 (impure)
-    if (value & SpecDoubleNaN)
+    // But we are not allowing such pure NaN pattern. Thus we can just follow to what
```

Comment:
Do we document that a `0xefff000000000000` isn't valid anywhere? Do we have assertions? This feels like a change that's likely to cause subtle breakages if we don't have assertions for tests/fuzzers to catch.

Nit: "But we don't allow such pure NaN patterns. Thus we don't need any special handling for those pure NaNs"

Reply:
One thing that might be worth a `static_assert` is that Clang/GCC always generate the pure NaN pattern we expect when constant folding.

Reply:
We already relying on this behavior for a long time because it is how JSValue is encoded. (We do not purifyNaN for any C++ double values with negation).
Also, we do not do anything in Baseline JIT etc.

Reply:
I changed isImpureNaN function to make it more explicit that it sees JSValue double as double.

Reply:
Ah, ok. IMO, the comment change is still worth it though.